client.py

```diff
@@ -105,20 +91,30 @@ def main():
     if 'StartdLogging' in config_dict:
         config_startd_logging = config_dict['StartdLogging']
     else:
-        config_startdlogging = {}
+        config_startd_logging = {}
+
+    # Loading secrets.  Fail if permissions wrong.
+    if oct(stat.S_IMODE(os.stat(options.secrets).st_mode)) != "0600":
```

Comment:
I wonder if there's a better way to do this than converting to a string

Reply:
I know it seems a bit icky.  I looked some more and couldn't find anything more straight forward.  You can use https://docs.python.org/2/library/stat.html to check RWX bits on Owner, Group and Everyone separately, but that would add 9 lines of code.  If you see anything else, let me know.  I would be happy to change it to something more readable.

Reply:
How about:
```python
if os.stat(options.secrets).st_mode & (stat.S_IXGRP | stat.S_IRWXO):
```
That tests for group or other permissions.

Reply:
Changed.